src/main/java/org/opentripplanner/graph_builder/module/NearbyStopFinder.java

```diff
@@ -80,34 +80,21 @@ public NearbyStopFinder(Graph graph, double radiusMeters, boolean useStreets) {
     }
 
     /**
-     * Find all unique nearby stops that are the closest stop on some trip pattern.
+     * Find all unique nearby stops
      * Note that the result will include the origin vertex if it is an instance of TransitStop.
      * This is intentional: we don't want to return the next stop down the line for trip patterns that pass through the
      * origin vertex.
      */
     public Set<StopAtDistance> findNearbyStopsConsideringPatterns (Vertex vertex) {
-
-        /* Track the closest stop on each pattern passing nearby. */
-        SimpleIsochrone.MinMap<TripPattern, StopAtDistance> closestStopForPattern =
-                new SimpleIsochrone.MinMap<TripPattern, StopAtDistance>();
+        Set<StopAtDistance> uniqueStops = Sets.newHashSet();
 
         /* Iterate over nearby stops via the street network or using straight-line distance, depending on the graph. */
         for (NearbyStopFinder.StopAtDistance stopAtDistance : findNearbyStops(vertex)) {
-            /* Filter out destination stops that are already reachable via pathways or transfers. */
-            // FIXME why is the above comment relevant here? how does the next line achieve this?
-            TransitStop ts1 = stopAtDistance.tstop;
-            if (!ts1.isStreetLinkable()) continue;
-            /* Consider this destination stop as a candidate for every trip pattern passing through it. */
-            for (TripPattern pattern : graph.index.patternsForStop.get(ts1.getStop())) {
-                closestStopForPattern.putMin(pattern, stopAtDistance);
+            if (stopAtDistance.tstop.isStreetLinkable()) {
+                uniqueStops.add(stopAtDistance);
```

Comment:
I think the idea here is to only have one stop included per route, even if multiple stops are close. Any idea why the shuttle trips aren't being found as a part of a pattern?

Reply:
Because, the stop, which we would like to link (`door-aport-buses`), doesn't have any trips going through it. The subway stop (`70048`) does have trips going through it but it's not "street linkable" (because it's linked to the street through the doors).

Reply:
To explain further it founds the `door-aport-buses` -> `2502248` (shuttle stop) connection using the original algorithm, but it doesn't find reverse `2502248` -> `door-aport-buses` connection because of what I have described in previous comment.

Reply:
What about checking for patterns, and only using them if there are any patterns for the stop? It sounds like we want to include entrances (which aren't on any routes), but we also don't want to double-include stops which are on a route but both nearby.

Reply:
We can do it, buy why don't we want to include multiple stops on the same route if they are all nearby (within 750 meters)? It may be 5 minutes walk to one stop and 6 minutes walk to another stop, but that other stop might be preferable because of the schedule (that is, you can catch the bus if you walk 6 minutes, but you're going to miss it if you walk 5 minutes to previous stop).

Reply:
Okay, let's give it a shot. I'm a little concerned that for plans where there are lots of stops (downtown for example) it'll end up being slow, but we can find out!

Reply:
The number of graph edges increases from **2284200** to **2399336** (+5%) because of this change, so in theory there should be no significant performance impact. We'll see though.